### PR TITLE
teleport.service: Secure SSH access via Teleport

### PR DESCRIPTION
**Pull request by tuladhar**
Towards
- https://github.com/giantswarm/roadmap/issues/2700
- https://github.com/giantswarm/roadmap/issues/3128

This PR adds
- teleport.service: Secure SSH access via Teleport

```
puru@giantswarm ~ % k get nodes -o wide
NAME                                   STATUS   ROLES           AGE     VERSION    INTERNAL-IP     EXTERNAL-IP     OS-IMAGE                                             KERNEL-VERSION     CONTAINER-RUNTIME
bigmac-bjhlc                           Ready    control-plane   2m14s   v1.24.12   10.10.222.103   10.10.222.103   Flatcar Container Linux by Kinvolk 3602.2.1 (Oklo)   5.15.136-flatcar   containerd://1.7.6
bigmac-qkmt6                           Ready    control-plane   6m45s   v1.24.12   10.10.222.50    10.10.222.50    Flatcar Container Linux by Kinvolk 3602.2.1 (Oklo)   5.15.136-flatcar   containerd://1.7.6
bigmac-shtjw                           Ready    control-plane   3m31s   v1.24.12   10.10.222.91    10.10.222.91    Flatcar Container Linux by Kinvolk 3602.2.1 (Oklo)   5.15.136-flatcar   containerd://1.7.6
bigmac-worker-6598747d7dxgx8dq-fcg7n   Ready    <none>          5m38s   v1.24.12   10.10.222.190   10.10.222.190   Flatcar Container Linux by Kinvolk 3602.2.1 (Oklo)   5.15.136-flatcar   containerd://1.7.6
bigmac-worker-6598747d7dxgx8dq-lr9vr   Ready    <none>          5m38s   v1.24.12   10.10.222.95    10.10.222.95    Flatcar Container Linux by Kinvolk 3602.2.1 (Oklo)   5.15.136-flatcar   containerd://1.7.6
bigmac-worker-6598747d7dxgx8dq-z68vc   Ready    <none>          5m40s   v1.24.12   10.10.222.112   10.10.222.112   Flatcar Container Linux by Kinvolk 3602.2.1 (Oklo)   5.15.136-flatcar   containerd://1.7.6
puru@giantswarm ~ % tsh ssh root@bigmac-bjhlc
Update Strategy: No Reboots
bigmac-bjhlc ~ # 
```

<details>
  <summary>How to run CI</summary>

Comment on this PR with:

```
/run cluster-test-suites
```
</details>

### CODE REVIEW

**Review by vxav — APPROVED** [review]
lgtm if it works

**Review by marians — COMMENTED** [review]
Looks like the indentation white space in your additions to `values.schema.json` is different from the rest (tabs vs. space)

File: helm/cluster-vsphere/values.schema.json [hunk 1]
```diff
@@ -84,6 +84,27 @@
...
+                    "title": "Teleport",
+                    "properties": {
+                        "proxy": {
+                            "type": "string",
+                            "title": "Teleport Proxy Address:Port",
+                            "default": "test.teleport.giantswarm.io:443"
+                        },
+                        "version": {
+                            "type": "string",
+                            "title": "Teleport Version",
```
Comment (marians) [inline helm/cluster-vsphere/values.schema.json:98]: Sentence case

```suggestion
                            "title": "Teleport version",
```

**Comment by tuladhar** [comment]
> Is this in fact a setting that you decided to expose to customers? Otherwise it might be better added to the `.internal` object.
> 

No, customer should not be aware of Teleport, it's for internal use only, so I agree to make it internal in that case.

**Comment by tuladhar** [comment]
> Does teleport only provide the proxy functionality? If yes, it might be worthwhile renaming `teleport` to `teleportProxy` and rename the `proxy` child property `address` or `host`.

No, teleport provides more than proxy functionality. And I checked the official teleport kube agent app, it uses `proxyAddr`, it makes sense to use that convention in here as well.
https://github.com/gravitational/teleport/blob/branch/v13/examples/chart/teleport-kube-agent/values.schema.json#L45C10-L45C19

**Review by marians — APPROVED** [review]

**Comment by erkanerol** [comment]
@tuladhar The PR for Flatcar is ready for review here https://github.com/giantswarm/cluster-vsphere/pull/98

The image I use there is built by our image builder pipeline and contains the `teleport` binary.

**Review by erkanerol — APPROVED** [review]
LGTM

**Pull request merged**